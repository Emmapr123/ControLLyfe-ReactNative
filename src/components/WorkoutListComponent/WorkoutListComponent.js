import React from 'react';
import { View, TouchableOpacity, StyleSheet } from 'react-native';
import { WorkoutTitleComponent } from '../WorkoutTitleComponent';
import { DeleteWorkoutButton } from '../../SVGs';

const WorkoutListComponent = ( props ) => {
  const {workouts = [], getWorkouts} = props
  
  const deleteWorkout = async(id) => {
    const res = await axios.delete(`http://localhost:3000/workout/${id}`)
  }
  
  return workouts.map((workout) =>
    <View key={workout._id} >
      <TouchableOpacity key={workout._id} onPress={ () => props.navigation.navigate("Show Workout", { id: workout._id }) } >
        <WorkoutTitleComponent key={workout._id} {...workout} />
      </TouchableOpacity>
      <TouchableOpacity
        onPress={() =>{
          deleteWorkout(workout._id)
          getWorkouts()
        }}>
      <DeleteWorkoutButton style={styles.deleteButton} color={'darkgrey'} width={25}/>
     </TouchableOpacity>
    </View>
  )
}

const styles = StyleSheet.create({
  
  deleteButton: {
    position: 'absolute', 
    bottom: 15, right:30, 
    shadowColor:'black', 
    shadowRadius: 8, 
    shadowOffset: {width: 5, height: 5}, 
    shadowOpacity: 0.3
  }
})

export { WorkoutListComponent }