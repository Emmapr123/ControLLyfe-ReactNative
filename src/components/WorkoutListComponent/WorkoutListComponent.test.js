import React from 'react';
import renderer from 'react-test-renderer';
import { WorkoutListComponent } from './WorkoutListComponent';

test('renders correctly', () => {
  const tree = renderer.create(<WorkoutListComponent />).toJSON();
  expect(tree).toMatchSnapshot();
});