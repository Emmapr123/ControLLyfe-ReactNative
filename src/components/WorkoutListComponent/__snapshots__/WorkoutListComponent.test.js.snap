// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
>
  <Text
    style={
      Object {
        "color": "darkgray",
        "fontSize": 32,
        "margin": 16,
      }
    }
  />
</RCTSafeAreaView>
`;
